File: logic/gserver/gameserver/src/i3k/gs/TCPFriendClient.java
// modified by ket.kio.RPCGen at Sat Dec 27 11:14:12 CST 2014.

package i3k.gs;

import ket.kio.PacketDecoder;
import ket.kio.PacketEncoder;
import ket.kio.TCPClient;
import ket.kio.SimplePacket;

import i3k.rpc.ABaseDencoder;
import i3k.rpc.Packet;

public class TCPFriendClient extends TCPClient<SimplePacket>
{

	public TCPFriendClient(RPCManager managerRPC)
	{
		super(managerRPC.getNetManager());
		this.managerRPC = managerRPC;
	}

	private class Dencoder extends ABaseDencoder
	{
		@Override
		public boolean doCheckPacketType(int ptype)
		{
			switch( ptype )
			{
			// friend to server
			case Packet.eF2SPKTStateAnnounce:
			case Packet.eF2SPKTSearchFriends:
			case Packet.eF2SPKTQueryFriendProp:
			case Packet.eF2SPKTSendMessageToFriend:
			case Packet.eF2SPKTCreateRole:
				return true;
			default:
				break;
			}
			return false;
		}
	}

	@Override
	public PacketEncoder<SimplePacket> getEncoder()
	{
		return dencoder;
	}

	@Override
	public PacketDecoder<SimplePacket> getDecoder()
	{
		return dencoder;
	}

	@Override
	public void onOpen()
	{
		managerRPC.onTCPFriendClientOpen(this);
	}

	@Override
	public void onOpenFailed(ket.kio.ErrorCode errcode)
	{
		managerRPC.onTCPFriendClientOpenFailed(this, errcode);
	}

	@Override
	public void onClose(ket.kio.ErrorCode errcode)
	{
		managerRPC.onTCPFriendClientClose(this, errcode);
	}

	@Override
	public void onPacketRecv(SimplePacket packet)
	{
		switch( packet.getType() )
		{
		// friend to server
		case Packet.eF2SPKTStateAnnounce:
			{
				Packet.F2S.StateAnnounce p = (Packet.F2S.StateAnnounce)packet;
				managerRPC.onTCPFriendClientRecvStateAnnounce(this, p);
			}
			break;
		case Packet.eF2SPKTSearchFriends:
			{
				Packet.F2S.SearchFriends p = (Packet.F2S.SearchFriends)packet;
				managerRPC.onTCPFriendClientRecvSearchFriends(this, p);
			}
			break;
		case Packet.eF2SPKTQueryFriendProp:
			{
				Packet.F2S.QueryFriendProp p = (Packet.F2S.QueryFriendProp)packet;
				managerRPC.onTCPFriendClientRecvQueryFriendProp(this, p);
			}
			break;
		case Packet.eF2SPKTSendMessageToFriend:
			{
				Packet.F2S.SendMessageToFriend p = (Packet.F2S.SendMessageToFriend)packet;
				managerRPC.onTCPFriendClientRecvSendMessageToFriend(this, p);
			}
			break;
		case Packet.eF2SPKTCreateRole:
			{
				Packet.F2S.CreateRole p = (Packet.F2S.CreateRole)packet;
				managerRPC.onTCPFriendClientRecvCreateRole(this, p);
			}
			break;
		default:
			break;
		}
	}

	//todo
	private Dencoder dencoder = new Dencoder();
	private RPCManager managerRPC;
}

File: logic/gclient/gclient/tcpfriendclient.cpp
// modified by ket.kio.RPCGen at Mon Jan 04 14:34:52 CST 2016.

#include "tcpfriendclient.h"
#include "rpcmanager.h"
#include "packet.h"

namespace I3K
{

	TCPFriendClient::TCPFriendClient(KET::KIO::NetManager &managerNet)
		: KET::KIO::TCPClient<KET::KIO::SimplePacket>(managerNet)
	{
	}

	bool TCPFriendClient::DoCheckPacketType(int ptype)
	{
		switch( ptype )
		{
		// friend to server
		case Packet::eF2SPKTStateAnnounce:
		case Packet::eF2SPKTSearchFriends:
		case Packet::eF2SPKTQueryFriendProp:
		case Packet::eF2SPKTSendMessageToFriend:
		case Packet::eF2SPKTCreateRole:
			return true;
		default:
			break;
		}
		return false;
	}

	void TCPFriendClient::OnOpen()
	{
		m_pRPCManager->AddTCPFriendClientOpenMessage();
	}

	void TCPFriendClient::OnOpenFailed(KET::KIO::ErrorCode errcode)
	{
		m_pRPCManager->AddTCPFriendClientOpenFailedMessage(errcode);
	}

	void TCPFriendClient::OnClose(KET::KIO::ErrorCode errcode)
	{
		m_pRPCManager->AddTCPFriendClientCloseMessage(errcode);
	}

	void TCPFriendClient::OnPacketRecv(const KET::KIO::SimplePacket *pPacket)
	{
		m_pRPCManager->AddTCPFriendClientPacket(pPacket);
	}
	void TCPFriendClient::OnPacketRecvParse(const KET::KIO::SimplePacket *pPacket)
	{
		switch( pPacket->GetType() )
		{
		case Packet::eF2SPKTStateAnnounce:
			{
				const Packet::F2S::StateAnnounce *p = dynamic_cast<const Packet::F2S::StateAnnounce*>(pPacket);
				m_pRPCManager->OnTCPFriendClientRecvStateAnnounce(this, p);
			}
			break;
		case Packet::eF2SPKTSearchFriends:
			{
				const Packet::F2S::SearchFriends *p = dynamic_cast<const Packet::F2S::SearchFriends*>(pPacket);
				m_pRPCManager->OnTCPFriendClientRecvSearchFriends(this, p);
			}
			break;
		case Packet::eF2SPKTQueryFriendProp:
			{
				const Packet::F2S::QueryFriendProp *p = dynamic_cast<const Packet::F2S::QueryFriendProp*>(pPacket);
				m_pRPCManager->OnTCPFriendClientRecvQueryFriendProp(this, p);
			}
			break;
		case Packet::eF2SPKTSendMessageToFriend:
			{
				const Packet::F2S::SendMessageToFriend *p = dynamic_cast<const Packet::F2S::SendMessageToFriend*>(pPacket);
				m_pRPCManager->OnTCPFriendClientRecvSendMessageToFriend(this, p);
			}
			break;
		case Packet::eF2SPKTCreateRole:
			{
				const Packet::F2S::CreateRole *p = dynamic_cast<const Packet::F2S::CreateRole*>(pPacket);
				m_pRPCManager->OnTCPFriendClientRecvCreateRole(this, p);
			}
			break;
		default:
			break;
		}
	}

}

